pkg/operator/ceph/csi/spec.go

```diff
@@ -683,3 +739,116 @@ func (r *ReconcileCSI) validateCSIVersion(ownerInfo *k8sutil.OwnerInfo) (*CephCS
+			holderPluginTolerations := getToleration(r.opConfig.Parameters, driver.toleration, pluginTolerations)
+			holderPluginNodeAffinity := getNodeAffinity(r.opConfig.Parameters, driver.nodeAffinity, pluginNodeAffinity)
+			// apply driver's plugin tolerations and node affinity
+			applyToPodSpec(&cephPluginHolder.Spec.Template.Spec, holderPluginNodeAffinity, holderPluginTolerations)
+
+			// apply resource request and limit from corresponding plugin container
+			applyResourcesToContainers(r.opConfig.Parameters, driver.resource, &cephPluginHolder.Spec.Template.Spec)
+
+			// Append the CEPH_CLUSTER_NAMESPACE env var so that the main container can use it to create the network
+			// namespace symlink to the Kubelet plugin directory
+			cephPluginHolder.Spec.Template.Spec.Containers[0].Env = append(
+				cephPluginHolder.Spec.Template.Spec.Containers[0].Env,
+				corev1.EnvVar{
+					Name:  "CEPH_CLUSTER_NAMESPACE",
+					Value: multusCluster.cluster.Namespace,
+				},
+			)
+
+			// Append the driver name so that the symlink file goes into the right location on the
+			// kubelet plugin directory
+			cephPluginHolder.Spec.Template.Spec.Containers[0].Env = append(
+				cephPluginHolder.Spec.Template.Spec.Containers[0].Env,
+				corev1.EnvVar{
+					Name:  "ROOK_CEPH_CSI_DRIVER_NAME",
+					Value: driver.fullName,
+				},
+			)
+
+			// Make the DS name unique per Ceph cluster
+			cephPluginHolder.Name = fmt.Sprintf("%s-%s", cephPluginHolder.Name, multusCluster.cluster.Name)
+			cephPluginHolder.Spec.Template.Name = fmt.Sprintf("%s-%s", cephPluginHolder.Spec.Template.Name, multusCluster.cluster.Name)
+			cephPluginHolder.Spec.Template.Spec.Containers[0].Name = fmt.Sprintf("%s-%s", cephPluginHolder.Spec.Template.Spec.Containers[0].Name, multusCluster.cluster.Name)
+
+			// Add default labels
+			k8sutil.AddRookVersionLabelToDaemonSet(cephPluginHolder)
+
+			// Apply Multus annotations to daemonset spec
+			err = k8sutil.ApplyMultus(multusCluster.cluster.Spec.Network, &cephPluginHolder.Spec.Template.ObjectMeta)
+			if err != nil {
+				return errors.Wrapf(err, "failed to apply multus configuration for holder %q in cluster %q", cephPluginHolder.Name, multusCluster.cluster.Namespace)
+			}
+
+			// Finally create the DaemonSet
+			_, err = r.context.Clientset.AppsV1().DaemonSets(r.opConfig.OperatorNamespace).Create(r.opManagerContext, cephPluginHolder, metav1.CreateOptions{})
+			if err != nil {
+				if kerrors.IsAlreadyExists(err) {
+					logger.Debugf("holder %q already exists for cluster %q, it should never be updated", cephPluginHolder.Name, multusCluster.cluster.Namespace)
+				} else {
+					return errors.Wrapf(err, "failed to start ceph plugin holder daemonset %q", cephPluginHolder.Name)
+				}
+			}
+
+			clusterConfigEntry := &CsiClusterConfigEntry{
+				Monitors: MonEndpoints(multusCluster.clusterInfo.Monitors),
+				RBD:      &CsiRBDSpec{},
+				CephFS:   &CsiCephFSSpec{},
+			}
+			if driver.name == RBDDriverShortName {
```

Comment:
Could you call `generateNetNamespaceFilePath()` once, then assign the same value in both places?